include/stl2/view/stride.hpp

```diff
@@ -0,0 +1,283 @@
+//
+// Project home: https://github.com/caseycarter/cmcstl2
+//
+#ifndef STL2_VIEW_STRIDE_HPP
+#define STL2_VIEW_STRIDE_HPP
+
+#include <stl2/detail/fwd.hpp>
+#include <stl2/detail/meta.hpp>
+#include <stl2/detail/iterator/concepts.hpp>
+#include <stl2/detail/non_propagating_cache.hpp>
+#include <stl2/detail/range/access.hpp>
+#include <stl2/detail/range/concepts.hpp>
+#include <stl2/detail/view/view_closure.hpp>
+#include <stl2/view/all.hpp>
+#include <stl2/view/view_interface.hpp>
+
+STL2_OPEN_NAMESPACE{
+	namespace ext {
+		template<input_range R>
+		requires view<R>
+		struct STL2_EMPTY_BASES stride_view
+		: view_interface<stride_view<R>> {
+		private:
+			template<bool>
+			struct __iterator;
+
+			using D = range_difference_t<R>;
+		public:
+			stride_view() = default;
+
+			constexpr explicit stride_view(R base, D stride)
+			: base_(static_cast<R&&>(base))
+			, stride_{stride}
+			{
+				STL2_EXPECT(stride_ > 0);
+			}
+
+			constexpr __iterator<false> begin() requires (!ext::simple_view<R>)
+			{ return __iterator<false>{*this}; }
+
+			constexpr __iterator<true> begin() const requires range<const R>
+			{ return __iterator<true>{*this}; }
+
+			constexpr auto end() requires (!ext::simple_view<R>)
+			{ return end_impl(*this); }
+
+			constexpr auto end() const requires range<const R>
+			{ return end_impl(*this); }
+
+			constexpr range_difference_t<R> size() requires (!ext::simple_view<R> && sized_range<R>)
+			{ return compute_distance(ranges::size(base_)); }
+
+			constexpr range_difference_t<R> size() const requires range<const R> && sized_range<R>
+			{ return compute_distance(ranges::size(base_)); }
+
+			constexpr range_difference_t<R> stride() const
+			{ return stride_; }
+
+			constexpr R base() const { return base_; }
+		private:
+			R base_{};
+			D stride_ = D{};
+
+			constexpr range_difference_t<R>
+			compute_distance(range_difference_t<R> const distance) const noexcept
+			{
+				auto const quotient = distance / stride_;
+				auto const remainder = distance % stride_;
+				return quotient + static_cast<range_difference_t<R>>(remainder > 0);
+			}
+
+			template<class Self>
+			static constexpr auto end_impl(Self& self)
+			{
+				if constexpr (sized_range<R> && common_range<R>) {
+					return __iterator<std::is_const_v<Self>>{
+						self,
+						__stl2::end(self.base_),
+						static_cast<range_difference_t<R>>(__stl2::size(self.base_)) % self.stride_
+					};
+				}
+				else {
+					return __stl2::end(self.base_);
+				}
+			}
+		};
+
+		template<range R>
+		stride_view(R&&, range_difference_t<R>) -> stride_view<all_view<R>>;
+
+		template<class R>
+		template<bool is_const>
+		struct stride_view<R>::__iterator {
+		private:
+			using parent_t = __maybe_const<is_const, stride_view>;
+			using base_t = __maybe_const<is_const, R>;
+
+
+			friend __iterator<!is_const>;
+
+			parent_t* parent_ = nullptr;
+			iterator_t<base_t> current_{};
+			range_difference_t<base_t> step_{};
```

Comment:
This should be a `non_propagating_cache`, but I've been having difficulty getting it to hold its value.